### DISCUSSION TITLE
Mock graphql-ws for tests (with jest, react-native)

### DISCUSSION THREAD

**Discussion by atosh502** [enisdenjo/graphql-ws#258] [General]
I am trying to setup tests for a react native app that uses graphql-ws for subscriptions. However, I am unsure how can I achieve this? Any directions would be very helpful. 

I think I need to mock the `createClient` method exported from `graphql-ws` but beyond that, I am clueless about how to do it. 

`import { createClient as createWSClient } from "graphql-ws";`

The following error appears when I run `yarn test`

![Screenshot from 2021-11-01 16-14-21](https://user-images.githubusercontent.com/19350071/139658278-20640f1b-f11f-40ba-839a-8c5f158be90a.png)

**Comment by enisdenjo** [comment]
Node does not have a WebSocket implementation in the global object, you must supply your own (which is what the error suggests). [`ws`](https://github.com/websockets/ws) is often used for Node environment, see ["Client usage in Node" recipe](https://github.com/enisdenjo/graphql-ws#node-client).

On how to test, I personally start a real server and connect to it as if it were any other. I dont mock requests because I want to test the whole transport and the library's behavior within. You can check my test suites under [`src/__tests__`](https://github.com/enisdenjo/graphql-ws/tree/master/src/__tests__), they might give you a few pointers.

**Reply by atosh502** [reply]
thank you I'll definitely check out the src/__tests__ directory...
